src/util/mod.rs

```diff
@@ -347,16 +430,75 @@ pub fn file_icon_suffix<P: AsRef<Path>>(f: P, is_file: bool) -> &'static str {
 ///
 /// The specified path must point to a file.
 pub fn get_raw_fs_metadata<P: AsRef<Path>>(f: P) -> RawFileData {
-    let f = f.as_ref();
+    get_raw_fs_metadata_impl(f.as_ref())
+}
+
+fn get_raw_fs_metadata_impl(f: &Path) -> RawFileData {
+    let meta = f.metadata().expect("Failed to get requested file metadata");
     RawFileData {
         mime_type: guess_mime_type_opt(f).unwrap_or_else(|| if file_binary(f) {
             "application/octet-stream".parse().unwrap()
         } else {
             "text/plain".parse().unwrap()
         }),
         name: f.file_name().unwrap().to_str().expect("Failed to get requested file name").to_string(),
-        last_modified: file_time_modified(f),
-        size: f.metadata().expect("Failed to get requested file metadata").len(),
+        last_modified: file_time_modified(&meta),
+        size: file_length(&meta, &f),
         is_file: true,
     }
 }
+
+/// Recursively copy a directory
+///
+/// Stolen from https://github.com/mdunsmuir/copy_dir/blob/0.1.2/src/lib.rs
+pub fn copy_dir(from: &Path, to: &Path) -> IoResult<Vec<(IoError, String)>> {
+    macro_rules! push_error {
+        ($vec:ident, $path:ident, $expr:expr) => {
+            match $expr {
+                Ok(_) => (),
+                Err(e) => $vec.push((e, $path.to_string_lossy().into_owned())),
+            }
+        };
+    }
+
+    let mut errors = Vec::new();
+
+    fs::create_dir(&to)?;
+
+    // The approach taken by this code (i.e. walkdir) will not gracefully
```

Comment:
Yeah, makes sense to detect and prevent something we aren't sure about. Is this a limitation that would be exposed to users to http? If so, can we put in an issues to capture the limitation and one day it can be fixed

Reply:
It'd be exposed by `COPY /owo` with `Destination: /owo/ewe`.

This itself is a problem with a wide array of unix tools – `cp` also refuses to copy a directory into itself, whereas `rsync` handles that gracefully.